pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,957 +71,765 @@
-        self._checkpoint_io: Optional[CheckpointIO] = None
-
-        plugins = plugins if plugins is not None else []
-
-        if isinstance(plugins, str):
-            plugins = [plugins]
-
-        if not isinstance(plugins, Sequence):
-            plugins = [plugins]
-
-        self.plugins = plugins
-
-        self._handle_accelerator_and_strategy()
-
-        self._validate_accelerator_and_devices()
-
-        self._warn_if_devices_flag_ignored()
-
-        self.select_accelerator_type()
-
-        if self._strategy_flag is not None:
-            self._set_strategy()
-        else:
-            self.set_distributed_mode()
-
-        self.handle_given_plugins()
-        self._set_strategy_type_if_strategy_passed()
-
-        self._cluster_environment = self.select_cluster_environment()
-
-        self.update_device_type_if_ipu_plugin()
-        self.update_device_type_if_strategy_passed()
-
-        self._validate_accelerator_type()
-        self._set_devices_if_none()
-
-        self.strategy = self.final_strategy()
-        self.accelerator = self.strategy.accelerator
-        self._check_plugin_compatibility()
-
-        # benchmarking
-        # TODO: should this be moved to GPU accelerator?
-        torch.backends.cudnn.benchmark = self.benchmark
+        """
+            A. accelerator flag could be:
+                1. strategy class (deprecated in 1.5 will be removed in 1.7)
+                2. strategy str (deprecated in 1.5 will be removed in 1.7)
+                3. accelerator class
+                4. accelerator str
+                5. accelerator auto
+
+            B. strategy flag could be :
+                1. strategy class
+                2. strategy str registered with strategyRegister
+                3. strategy str in _strategy_type enum which listed in each strategy as
+                   backend (registed these too, and _strategy_type could be deprecated)
+
+            C. plugins flag could be:
+                1. List of str, which could contains:
+                    i. strategy str
+                    ii. precision str (Not supported in the old accelerator_connector version)
+                    iii. checkpoint_io str (Not supported in the old accelerator_connector version)
+                    iv. cluster_environment str (Not supported in the old accelerator_connector version)
+                2. List of class, which could contains:
+                    i. strategy class (deprecated in 1.5 will be removed in 1.7)
+                    ii. precision class (should be removed, and precision flag should allow user pass classes)
+                    iii. checkpoint_io class
+                    iv. cluster_environment class
+
+
+        priorities which to take when:
+            A. Class > str
+            B. Strategy > Accelerator/precision/plugins
+            C. When multiple flag set to the same thing? (ignore? not handled for now)
 
+        """
+        torch.backends.cudnn.benchmark = benchmark
```

Comment:
```suggestion
        # TODO: move to gpu accelerator
        torch.backends.cudnn.benchmark = benchmark
```

still valid, I think :)